### PR TITLE
#11291 add scroll ability to ssh dialog

### LINKED ISSUE #11921: Test tunnel configuration button cut off
#### System information: 
- Ubuntu 20.10 64-bit Gnome 3.38.3 X11
- DBeaver 21.0.1

#### Describe the problem you're observing:
When creating a MariaDB connection using SSH Tunnel the button **Test tunnel configuration** is not visible because the default height of the window is to small. You can resize the window but that isn't obvious for the user and the window should have a different height by default.

Screenshot from 2021-03-30 11-29-42

### PR DESCRIPTION
**Pull request by Destrolaric**
Closes #11921
image

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.net.ssh.ui/src/org/jkiss/dbeaver/ui/net/ssh/SSHTunnelConfiguratorUI.java [hunk 1]
```diff
@@ -103,6 +109,7 @@ public void createControl(Composite parent, Object object, Runnable propertyChan
                 @Override
                 public void expansionStateChanged(ExpansionEvent e) {
                     UIUtils.resizeShell(parent.getShell());
+                    scrolledComposite.setMinSize(parent.computeSize(SWT.DEFAULT, SWT.DEFAULT));
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.net.ssh.ui/src/org/jkiss/dbeaver/ui/net/ssh/SSHTunnelConfiguratorUI.java]: I believe this must be done before resizing the shell.

File: plugins/org.jkiss.dbeaver.net.ssh.ui/src/org/jkiss/dbeaver/ui/net/ssh/SSHTunnelConfiguratorUI.java [hunk 2]
```diff
@@ -87,11 +88,16 @@ public class SSHTunnelConfiguratorUI implements IObjectPropertyConfigurator<Obje
...
+        ScrolledComposite scrolledComposite = new ScrolledComposite(parent, SWT.V_SCROLL);
+        scrolledComposite.setLayout(new GridLayout(1, false));
+        scrolledComposite.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+        final Composite composite = new Composite(scrolledComposite, SWT.NONE);
         composite.setLayoutData(new GridData(GridData.FILL_BOTH));
         composite.setLayout(new GridLayout(1, false));
-
+        scrolledComposite.setContent(composite);
+        scrolledComposite.setExpandHorizontal( true );
+        scrolledComposite.setExpandVertical( true );
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.net.ssh.ui/src/org/jkiss/dbeaver/ui/net/ssh/SSHTunnelConfiguratorUI.java]: No spaces around method arguments.

File: plugins/org.jkiss.dbeaver.net.ssh.ui/src/org/jkiss/dbeaver/ui/net/ssh/SSHTunnelConfiguratorUI.java [hunk 3]
```diff
@@ -87,11 +88,16 @@ public class SSHTunnelConfiguratorUI implements IObjectPropertyConfigurator<Obje
     @Override
     public void createControl(Composite parent, Object object, Runnable propertyChangeListener)
     {
-        final Composite composite = new Composite(parent, SWT.NONE);
-        //gd.minimumHeight = 200;
+        ScrolledComposite scrolledComposite = new ScrolledComposite(parent, SWT.V_SCROLL);
+        scrolledComposite.setLayout(new GridLayout(1, false));
+        scrolledComposite.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+        final Composite composite = new Composite(scrolledComposite, SWT.NONE);
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.net.ssh.ui/src/org/jkiss/dbeaver/ui/net/ssh/SSHTunnelConfiguratorUI.java:95]: Please add an extra line to visually separate declarations.

File: plugins/org.jkiss.dbeaver.net.ssh.ui/src/org/jkiss/dbeaver/ui/net/ssh/SSHTunnelConfiguratorUI.java [hunk 4]
```diff
@@ -183,12 +191,6 @@ public void widgetSelected(SelectionEvent e) {
             setNumberEditStyles(tunnelTimeout);
         }
 
-        {
-            Composite sc = new Composite(composite, SWT.NONE);
-            sc.setLayout(new GridLayout());
-            sc.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
-        }
-
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.net.ssh.ui/src/org/jkiss/dbeaver/ui/net/ssh/SSHTunnelConfiguratorUI.java]: It was a spacer that forced the link widget to be at the very bottom. What's the point of this change?

**Review by ShadelessFox — CHANGES_REQUESTED** [review]
Please attach screenshots showing visual changes.

**Review by ShadelessFox — APPROVED** [review]

**Comment by Matvey16** [comment]
Scroll bar stays active and shows empty space when settings are collapsed back
VirtualBoxVM_bXkEmw0LXD

**Comment by Destrolaric** [comment]
Scrollable composite works badly with resizeShell, it can do it vertically but not horizontally, so I hardcoded shell width.

**Comment by Destrolaric** [comment]
Fully shown:
image
Folded: 
image

**Comment by Matvey16** [comment]
Verified

**Pull request merged**